### PR TITLE
Add unit tests for missing required attributes in population methods

### PR DESCRIPTION
**Pull request by langner**
This is a paraphrase and extension of #480.

### CODE REVIEW

File: cclib/method/population.py [hunk 1]
```diff
@@ -32,6 +34,13 @@ def __repr__(self):
         """Return a representation of the object."""
         return "Population"
 
+    def _check_required_attributes(self):
+        super(Population, self)._check_required_attributes()
+        
+        if not any(hasattr(self.data, a) for a in self.overlap_attributes):
+            raise MissingAttributeError(
+                    'Need overlap matrix (aooverlaps or fooverlaps attribute) for Population methods')
```
Comment (ATenderholt) [inline cclib/method/population.py]: The overlap matrix isn't required for CSPA.

File: cclib/method/population.py [hunk 2]
```diff
@@ -11,17 +11,19 @@
...
 
-from cclib.method.calculationmethod import Method
+from cclib.method.calculationmethod import Method, MissingAttributeError
 
 
 class Population(Method):
     """An abstract base class for population-type methods."""
 
+    required_attrs = ('homos', 'mocoeffs', 'nbasis')
+    overlap_attributes = ("aooverlaps", "fooverlaps")
```
Comment (ATenderholt) [inline cclib/method/population.py]: Any reason you switched from single quotes to double quotes? (No real preference, just was surprised.)

File: cclib/method/population.py (hunk 1 above)
Comment (langner) [reply cclib/method/population.py]: Indeed! Done in 83c0738ef18535f16e807a1f5235329e54cf6a3b.

File: cclib/method/population.py (hunk 2 above)
Comment (langner) [reply cclib/method/population.py]: Done in 83c0738ef18535f16e807a1f5235329e54cf6a3b.

**Pull request merged**